libraries/rush-lib/src/pluginFramework/RushLifeCycle.ts

```diff
@@ -35,6 +36,12 @@ export interface IPhasedCommand extends IRushCommand {
    * @alpha
    */
   readonly hooks: PhasedCommandHooks;
+
+  /**
+   * Function that return current phase command result
+   * @alpha
+   */
+  readonly getCurrentExecutionResult: () => IExecutionResult['operationResults'] | undefined;
```

Comment:
Would rather add hooks to `PhasedCommandHooks` to give you the records immediately after initialization, and to notify you when the state of a record changes.

Reply:
yeah, i agree this is a better solution to achieve this requirement.